src/wp-includes/ms-functions.php

```diff
@@ -1016,7 +1017,7 @@ function wpmu_signup_blog_notification(
 			$meta
 		),
 		$activate_url,
-		esc_url( "http://{$domain}{$path}" ),
+		esc_url( ( is_ssl() ? 'https://' : 'http://' ) . $domain . $path ),
```

Comment:
Use variable and then use it in esc_url() 
`$scheme = is_ssl() ? 'https://' : 'http://';`